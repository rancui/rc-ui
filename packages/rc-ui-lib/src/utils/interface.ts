import type { CSSProperties } from 'react';

export interface BaseTypeProps {
  style?: CSSProperties;
  className?: string;
  children?: React.ReactNode;
}

/** 指定挂载的节点 */
export type TeleportType = HTMLElement | (() => HTMLElement) | null;
/** 带有displayName的组件 */
export type WithDisplayNameReactElement = React.ReactElement & { type: { displayName: string } };
