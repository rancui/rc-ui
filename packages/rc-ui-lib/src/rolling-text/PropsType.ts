import { BaseTypeProps } from '../utils';

export type RollingTextDirection = 'up' | 'down';
export type RollingTextStopOrder = 'ltr' | 'rtl';

export interface RollingTextProps extends BaseTypeProps {
  /**
   * 数字高度，单位为 px
   * @default 40
   */
  height?: number;
  /**
   * 起始数值
   * @default 0
   */
  startNum?: number;
  /**
   * 目标数值
   */
  targetNum?: number;
  /**
   * 内容数组，用于翻转非数字内容
   * @default []
   */
  textList?: string[];
  /**
   * 动画时长，单位为秒
   * @default 2
   */
  duration?: number;
  /**
   * 是否自动开始动画
   * @default true
   */
  autoStart?: boolean;
  /**
   * 文本翻滚方向，值为 down 和 up
   * @default down
   */
  direction?: RollingTextDirection;
  /**
   * 各个数位动画停止先后顺序，值为 ltr 和 rtl
   * @default ltr
   */
  stopOrder?: RollingTextStopOrder;
}

export type RollingTextInstance = {
  start: () => void;
  reset: () => void;
};

export interface RollingTextItemProps {
  figureArr?: any[];
  /**
   * 起始数值
   * @default 0
   */
  delay?: number;
  /**
   * 动画时长，单位为秒
   * @default 2
   */
  duration?: number;
  /**
   * 是否自动开始动画
   * @default true
   */
  isStart: boolean;
  /**
   * 文本翻滚方向，值为 down 和 up
   * @default down
   */
  direction?: RollingTextDirection;
  /**
   * 数字高度，单位为 px
   * @default 40
   */
  height?: number;
}

export type RollingTextThemeVars = {
  rollingTextBackground?: string;
  rollingTextColor?: string;
  rollingTextFontSize?: string;
  rollingTextGap?: string;
  rollingTextItemWidth?: string;
  rollingTextItemBorderRadius?: string;
};
