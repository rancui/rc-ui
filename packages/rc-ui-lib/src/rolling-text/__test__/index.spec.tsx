import React from 'react';
import { act, cleanup, render } from '@testing-library/react';
import { RollingText, type RollingTextProps } from '..';
import { RollingTextInstance } from '../PropsType';
import { sleep } from '../../../tests/utils';

const itemWrapperClass = '.rc-rolling-text-item__box';
const animationClass = 'rc-rolling-text-item__box--animate';

describe('RollingText test', () => {
  function createRollingText(props: RollingTextProps) {
    const rollingTextRef = React.createRef<RollingTextInstance>();
    const { container, rerender, debug } = render(<RollingText ref={rollingTextRef} {...props} />);

    return {
      rollingTextRef,
      container,
      rerender,
      debug,
    };
  }

  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should render comp', () => {
    const { container } = createRollingText({
      startNum: 0,
      targetNum: 123,
    });

    expect(container).toMatchSnapshot();
  });

  it('should render correctly when using textList prop', () => {
    const { container } = createRollingText({
      textList: ['aaaaa', 'bbbbb', 'ccccc', 'ddddd', 'eeeee', 'fffff', 'ggggg'],
      duration: 1,
    });

    expect(container).toMatchSnapshot();
  });

  test('should start rolling when auto-start prop is true', async () => {
    const { container } = createRollingText({
      startNum: 0,
      targetNum: 123,
      autoStart: true,
      stopOrder: 'rtl',
    });

    expect(container.querySelector(itemWrapperClass).classList).toContain(animationClass);
  });

  test('should not start rolling when auto-start prop is false', async () => {
    const { container } = createRollingText({
      startNum: 0,
      targetNum: 123,
      autoStart: false,
      direction: 'up',
    });

    expect(container.querySelector(itemWrapperClass).classList).not.toContain(animationClass);
  });

  test('should start rolling after calling the start method', async () => {
    const { container, rollingTextRef } = createRollingText({
      startNum: 0,
      targetNum: 123,
      autoStart: false,
    });

    act(() => {
      rollingTextRef.current.start();
    });
    await sleep();
    expect(container.querySelector(itemWrapperClass).classList).toContain(animationClass);
  });

  test('should reset the animation after calling the reset method', async () => {
    const { container, rollingTextRef } = createRollingText({
      startNum: 0,
      targetNum: 123,
      autoStart: false,
    });

    act(() => {
      rollingTextRef.current.start();
    });
    await sleep();
    expect(container.querySelector(itemWrapperClass).classList).toContain(animationClass);

    act(() => {
      rollingTextRef.current.reset();
    });
    await sleep();
    expect(container.querySelector(itemWrapperClass).classList).not.toContain(animationClass);
  });

  test('should restart rolling after calling the reset method when auto-start prop is true', async () => {
    const { container, rollingTextRef } = createRollingText({
      startNum: 0,
      targetNum: 123,
      autoStart: true,
    });

    act(() => {
      rollingTextRef.current.reset();
    });
    await sleep();
    expect(container.querySelector(itemWrapperClass).classList).not.toContain(animationClass);
    await sleep(50);
    expect(container.querySelector(itemWrapperClass).classList).toContain(animationClass);
  });
});
