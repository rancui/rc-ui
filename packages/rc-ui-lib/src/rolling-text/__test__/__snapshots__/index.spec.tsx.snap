// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RollingText test should render comp 1`] = `
<div>
  <div
    class="rc-rolling-text"
  >
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -1640px; --rc-duration: 2s; --rc-delay: 0s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
      </div>
    </div>
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -1680px; --rc-duration: 2s; --rc-delay: 0.2s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
      </div>
    </div>
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -1720px; --rc-duration: 2s; --rc-delay: 0.4s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          9
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          8
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          7
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          6
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          5
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          4
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          3
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          2
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          1
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RollingText test should render correctly when using textList prop 1`] = `
<div>
  <div
    class="rc-rolling-text"
  >
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -240px; --rc-duration: 1s; --rc-delay: 0s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          g
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          f
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          e
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          d
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          c
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          b
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          a
        </div>
      </div>
    </div>
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -240px; --rc-duration: 1s; --rc-delay: 0.2s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          g
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          f
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          e
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          d
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          c
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          b
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          a
        </div>
      </div>
    </div>
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -240px; --rc-duration: 1s; --rc-delay: 0.4s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          g
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          f
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          e
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          d
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          c
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          b
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          a
        </div>
      </div>
    </div>
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -240px; --rc-duration: 1s; --rc-delay: 0.6000000000000001s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          g
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          f
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          e
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          d
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          c
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          b
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          a
        </div>
      </div>
    </div>
    <div
      class="rc-rolling-text-item rc-rolling-text-item--down"
      style="height: 40px; --rc-translate: -240px; --rc-duration: 1s; --rc-delay: 0.8s;"
    >
      <div
        class="rc-rolling-text-item__box rc-rolling-text-item__box--animate"
      >
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          g
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          f
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          e
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          d
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          c
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          b
        </div>
        <div
          class="rc-rolling-text-item__item"
          style="line-height: 40px;"
        >
          a
        </div>
      </div>
    </div>
  </div>
</div>
`;
